ftplugin/lua.lua

```diff
@@ -0,0 +1,86 @@
+local fmt = string.format
+
+-- Iterator that splits a string o a given delimiter
+local function split(str, delim)
+  delim = delim or "%s"
+  return string.gmatch(str, fmt("[^%s]+", delim))
+end
+
+-- Find the proper directory separator depending
+-- on lua installation or OS.
+local function dir_separator()
+  -- Look at package.config for directory separator string (it's the first line)
+  if package.config then
+    return string.match(package.config, "^[^\n]")
+  elseif vim.fn.has "win32" == 1 then
+    return "\\"
+  else
+    return "/"
+  end
+end
+
+-- Search for lua traditional include paths.
+-- This mimics how require internally works.
+local function include_paths(fname, ext)
+  ext = ext or "lua"
+  local sep = dir_separator()
+  local paths = string.gsub(package.path, "%?", fname)
+  for path in split(paths, "%;") do
+    if vim.fn.filereadable(path) == 1 then
+      return path
+    end
+  end
+end
+
+-- Search for nvim lua include paths
+local function include_rtpaths(fname, ext)
+  ext = ext or "lua"
+  local sep = dir_separator()
```

Comment:
@SingularisArt, please fix the lint violations here, these two variables are not used